machine-runner/src/runner/runner.ts

```diff
@@ -525,14 +530,17 @@ export const createMachineRunnerInternal = <
       cloneFrom: state,
     })
 
+    const purgeWhenMatching = ({ sourceState }: { sourceState: S }) =>
+      nva.purgeWhenMatching(sourceState)
+
     emitter.on('next', nva.push)
     emitter.on('failure', nva.fail)
-    emitter.on('commandPersisted', nva.purge)
+    emitter.on('commandPersisted', purgeWhenMatching)
```

Comment:
why do we do this at all?

Reply:
This started when we observed, during one of Jose's presentation on SW's ActyxServiceBase refactor, that a command call should purge so that subsequent `.next`/`.peekNext` call will not be the same state that calls the command.

As for the specific timing of the purge (on persistence), this is the reasoning https://github.com/Actyx/machines/pull/92#discussion_r1466491054